glusterfs/datadog_checks/glusterfs/check.py

```diff
@@ -0,0 +1,207 @@
+
+        # Check if customer set gstatus path
+        if init_config.get('gstatus_path'):
+            self.gstatus_cmd = init_config.get('gstatus_path')
+        else:
+            if os.path.exists(GSTATUS_PATH):
+                self.gstatus_cmd = GSTATUS_PATH
+            elif os.path.exists(INSTALL_PATH):
+                self.gstatus_cmd = INSTALL_PATH
+            else:
+                raise ConfigurationError(
+                    'Glusterfs check requires `gstatus` to be installed or set the path to the installed version.'
+                )
+        self.log.debug("Using gstatus path `%s`", self.gstatus_cmd)
+
+    def check(self, _):
+        use_sudo = _is_affirmative(self.instance.get('use_sudo', False))
+        if use_sudo:
+            test_sudo = os.system('setsid sudo -l < /dev/null')
+            if test_sudo != 0:
+                raise Exception('The dd-agent user does not have sudo access')
+            gluster_args = ['sudo', self.gstatus_cmd]
+        else:
+            gluster_args = [self.gstatus_cmd]
+
+        # Ensures units are universally the same by specifying the --units flag
+        gluster_args += ['-a', '-o', 'json', '-u', 'g']
+        self.log.debug("gstatus command: %s", gluster_args)
+        output, _, _ = get_subprocess_output(gluster_args, self.log)
+        gstatus = json.loads(output)
+
+        if 'data' in gstatus:
+            data = gstatus['data']
+            self.submit_metric(data, 'cluster', CLUSTER_STATS, self._tags)
+
+            self.submit_version_metadata(data)
+
+            if 'volume_summary' in output:
+                self.parse_volume_summary(data['volume_summary'])
+
+            if CLUSTER_STATUS in data:
+                status = data[CLUSTER_STATUS]
+                if status == 'Healthy':
+                    self.service_check(self.CLUSTER_SC, AgentCheck.OK, tags=self._tags)
+                else:
+                    self.service_check(
+                        self.CLUSTER_SC, AgentCheck.CRITICAL, tags=self._tags, message="Cluster status is %s" % status
+                    )
+        else:
+            self.log.warning("No data from gstatus")
+
+    @AgentCheck.metadata_entrypoint
+    def submit_version_metadata(self, data):
+        try:
+            raw_version = data[GLUSTER_VERSION]
+        except KeyError as e:
+            self.log.debug("Could not retrieve GlusterFS version: %s", str(e))
+
+        if raw_version:
+            version_parts = self.parse_version(raw_version)
+
+            self.set_metadata('version', raw_version, scheme='parts', part_map=version_parts)
+            self.log.debug('Found GlusterFS version: %s', raw_version)
+        else:
+            self.log.warning('Could not retrieve GlusterFS version info: %s', raw_version)
+
+    def parse_version(self, version):
+        """
+        GlusterFS versions are in format <major>.<minor>
+        """
+        major, minor = version.split('.')
```

Comment:
This will trigger an exception if the version format is unexepected. We try to be extra-careful with metadata submission, let's make sure that an exception in metadata collection does not stop the check from working.